### ISSUE TITLE
Add mate length as a complexity term for difficulty calculation

### ISSUE THREAD

**Issue by Belzedar94** [ianfab/chess-variant-puzzler#15]
Below two puzzles share the same difficulty while the second one is significantly harder than the first one. Does it make sense to reflect mate length as a factor in the difficulty calculation if puzzle type = mate?

`1nbqkbn1/1ppppRp1/5r2/p3R2p/P6P/8/1PPPPrP1/1NBQKBN1 w - - 1 9;variant kamikazerooks;bm e5e7;eval #2;difficulty 6;quality 1.00;type mate;pv e5e7,f8e7,f7f8`

`3k2B1/1p1bR1p1/2p5/p5P1/P2p2rP/1P1P4/3N4/1N2K3 w - - 2 26;variant kamikazerooks;bm e7e8;eval #7;difficulty 6;quality 0.73;type mate;pv e7e8,d8c7,e8c8,c7d6,d2c4,d6c5,c8c6,c5b4,c6b6`

**Comment by ianfab** [comment]
Yes, let's collect patterns about the difficulty evaluation here and then after some time we can see if we can come up with a better definition.
